docs/migration.rst

```diff
@@ -4,36 +4,93 @@ Migrating from catkin_make
+Implications of Isolation
+--------------------------
+
+Build isolation has the following implications for both ``catkin_make_isolated`` and ``catkin build``:
+
+- There is no "top-level" ``CMakeLists.txt`` file in the **source space**.
 - Each package in a ``catkin_tools`` workspace has its own isolated build space.
-- ``catkin build`` can be run from any directory under the workspace root.
-- ``catkin config`` stores many workspace configuration options which needed to be passed to each call of ``catkin_make``.
-- ``catkin build`` can build plain CMake packages if they have ``package.xml`` files.
 - Packages built with ``catkin build`` can not access variables defined in other Catkin packages in the same workspace.
-- Packages no longer need to define target dependencies on ROS messages built in other packages.
-  All targets in a dependency are guaranteed to have been   built before the current package.
-- ``catkin_tools`` and ``catkin_make`` can use the same source space, but they must use different build, devel, and install spaces.
-- ``catkin build`` generates ``.catkin`` files and subsequently ``ROS_PACKAGE_PATH`` variables where each source package is listed, individually,   instead of just listing the source space for the workspace.
-- ``catkin build`` passes CMake command line arguments to multiple packages.
-  Since not all packages accept the same CMake arguments, the ``cmake`` command is invoked with ``--no-warn-unused-cli``.
-  This means there will be no warnings for unused variables passed to ``cmake``.
+- All targets in each of a package's dependencies are guaranteed to have been built before the current package.
+- Packages do not need to define target dependencies on ROS messages built in other packages.
+- Plain CMake packages can be built if they each have a ``package.xml`` file with the appropriate `<build_type> tag <http://www.ros.org/reps/rep-0140.html#build-type>`_.
 
-IDE Integration
-^^^^^^^^^^^^^^^
+Additional Differences with ``catkin build``
+--------------------------------------------
 
-Since all packages are built in isolation with ``catkin build``, you can't rely on CMake's IDE integration to generate a single project for your entire workspace.
+In addition to the differences due to isolation, ``catkin build`` is also different from ``catkin_make_isolated`` in the following ways:
+
+- It builds packages in parallel, using an internal job server to distribute load.
+- It puts products into hidden directories, and then symbolically links them into the **devel space**.
+- It stores persistent configuration options in a ``.catkin_tools`` directory at the root of your workspace.
+- It passes the same CMake command line arguments to multiple packages.
+- It passes ``--no-warn-unused-cli`` to the ``cmake`` command since not all packages accept the same CMake arguments.
+- It generates ``.catkin`` files where each source package is listed, individually, instead of just listing the source space for the workspace.
```

Comment:
This could do with an example like:

> e.g. the `.catkin` file for `catkin_make_isolated`  might contain a single line like `/path/to/my_ws/src` whereas `catkin build`'s `.catkin` will contain a line for each package, each like `/path/to/my_ws/src/<package source folder>`.

And it might link to the issues about this:

- https://github.com/catkin/catkin_tools/issues/221
- https://github.com/catkin/catkin_tools/issues/283

Or if you don't link the issue here, do a reverse link to the documentation from the issue so we remember to change this doc if we change the behavior.